File: app/src/main/java/com/mab/studentrequest/LoginActivity.java
package com.mab.studentrequest;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

import java.util.regex.Pattern;

public class LoginActivity extends AppCompatActivity {
    private Button loginButton, registerButton;
    private EditText email, password;
    private View loader;
    private FirebaseAuth firebaseAuth;
    private String TAG = LoginActivity.class.getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        setIds();
        setListeners();
    }

    @Override
    protected void onStart() {
        super.onStart();
        if (firebaseAuth.getCurrentUser() != null)
            gotoMainActivity();
    }

    private void setIds() {
        loginButton = findViewById(R.id.login_button);
        registerButton = findViewById(R.id.register_button);
        email = findViewById(R.id.email);
        password = findViewById(R.id.password);
        loader = findViewById(R.id.loader);
        firebaseAuth = FirebaseAuth.getInstance();
    }


    private void setListeners() {
        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (performValidation()) {
                    loginUser();
                }
            }
        });

        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
            }
        });
    }


    private boolean performValidation() {
        if (!isEmailValid(email.getText().toString())) {
            email.requestFocus();
            email.setError("Enter valid email");
            return false;
        }
        if (password.getText().toString().isEmpty() || password.getText().toString().length() < 6) {
            password.requestFocus();
            password.setError("Enter valid password min 6 char");
            return false;
        }
        return true;
    }


    private void gotoMainActivity() {
        startActivity(new Intent(this, MainActivity.class));
    }


    private void loginUser() {
        loader.setVisibility(View.VISIBLE);
        firebaseAuth.signInWithEmailAndPassword(email.getText().toString(), password.getText().toString())
                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        loader.setVisibility(View.GONE);
                        if (task.isSuccessful()) {
                            Log.d(TAG, "signInWithEmail:success");
                            gotoMainActivity();
                        } else {
                            Log.w(TAG, "signInWithEmail:failure", task.getException());
                            Toast.makeText(LoginActivity.this, task.getException().getMessage(), Toast.LENGTH_SHORT).show();
                        }
                    }
                });

    }

    private boolean isEmailValid(String email) {
        Pattern emailPattern = Pattern.compile("[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]{2}+[a-z]*");
        return emailPattern.matcher(email).matches() && !email.trim().isEmpty();
    }

}

File: app/src/main/java/com/mab/studentrequest/LoginActivity.kt
package com.mab.studentrequest

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : AppCompatActivity() {

    private val TAG = LoginActivity::class.java.simpleName
    private val firebaseAuth by lazy {
        FirebaseAuth.getInstance()
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        setListeners()
    }

    public override fun onStart() {
        super.onStart()
        if (firebaseAuth.currentUser != null)
            gotoMainActivity()
    }

    private fun setListeners() {
        login_button.setOnClickListener {
            if (performValidation()) {
                loginUser()
            }
        }

        register_button.setOnClickListener {
            startActivity(Intent(this, RegisterActivity::class.java))
        }
    }

    private fun performValidation(): Boolean {
        if (!email.text.toString().isEmailValid()) {
            email.requestFocus()
            email.error = "Enter valid email"
            return false
        }
        if (password.text.isBlank() || password.text.length < 6) {
            password.requestFocus()
            password.error = "Enter valid password min 6 char"
            return false
        }
        return true
    }


    private fun gotoMainActivity() {
        startActivity(Intent(this, MainActivity::class.java))
    }


    private fun loginUser() {
        loader.visibility = View.VISIBLE
        firebaseAuth.signInWithEmailAndPassword(email.text.toString(), password.text.toString())
            .addOnCompleteListener(this) { task ->
                loader.visibility = View.GONE
                if (task.isSuccessful) {
                    Log.d(TAG, "signInWithEmail:success")
                    val user = firebaseAuth.currentUser
                    gotoMainActivity()
                } else {
                    Log.w(TAG, "signInWithEmail:failure", task.exception)
                    Toast.makeText(baseContext, task.exception?.message, Toast.LENGTH_SHORT).show()
                }
            }
    }
}
